member.php

```diff
@@ -125,7 +125,14 @@
 
 	if($mybb->settings['regtype'] == "randompass")
 	{
-		$mybb->input['password'] = random_str();
+		$mybb->settings['requirecomplexpasswords'] = 0;
+
+		$password_length = (int)$mybb->settings['minpasswordlength'];
+		if($password_length < 8)
+		{
+			$password_length = 8;
+		}
+		$mybb->input['password'] = random_str($password_length);

```

Comment:
This does not cover the rare but possible case having maxpasswordlength lower than 8 (the generated password being to long).